services/workflows-service/src/common/guards/combined-auth.guard.ts

```diff
@@ -0,0 +1,84 @@
+import type { Request } from 'express';
+import { ClsService } from 'nestjs-cls';
+import { CanActivate, ExecutionContext, Injectable, UnauthorizedException } from '@nestjs/common';
+
+import { AuthenticatedEntity, TProjectIds } from '@/types';
+import { WorkflowService } from '@/workflow/workflow.service';
+import { WorkflowTokenService } from '@/auth/workflow-token/workflow-token.service';
+
+@Injectable()
+export class CombinedAuthGuard implements CanActivate {
```

Comment:
what do you think about
```suggestion
export class WorkflowAuthGuard implements CanActivate {
```
as this seems very coupled to authenticating access to workflows.